src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java

```diff
@@ -0,0 +1,288 @@
+        }
+        argList.add("--address");
+        argList.add(ipAddress);
+
+        File log = getLogFile();
+        if (log != null){
+            argList.add("--log");
+            argList.add(log.getAbsolutePath());
+        }
+
+        Set<Map.Entry<String, String>> entries = serverArguments.entrySet();
+        Iterator<Map.Entry<String, String>> iterator = entries.iterator();
+        while (iterator.hasNext()){
+            Map.Entry<String, String> entry = iterator.next();
+            String argument = entry.getKey();
+            String value = entry.getValue();
+            if (StringUtils.isBlank(argument) || StringUtils.isBlank(value))
+                continue;
+
+            argList.add(argument);
+            argList.add(value);
+        }
+
+        ImmutableList<String> result = new ImmutableList.Builder<String>().addAll(argList).build();
+        return result;
+    }
+
+    /**
+     * Sets which Node.js the builder will use.
+     *
+     * @param nodeJSExecutable The executable Node.js to use.
+     * @return A self reference.
+     */
+    public AppiumServiceBuilder usingDriverExecutable(File nodeJSExecutable) {
+        return super.usingDriverExecutable(nodeJSExecutable);
+    }
+
+    /**
+     * Sets which port the appium server should be started on. A value of 0 indicates that any
+     * free port may be used.
+     *
+     * @param port The port to use; must be non-negative.
+     * @return A self reference.
+     */
+    public AppiumServiceBuilder usingPort(int port) {
+        return super.usingPort(port);
+    }
+
+    /**
+     * Configures the appium server to start on any available port.
+     *
+     * @return A self reference.
+     */
+    public AppiumServiceBuilder usingAnyFreePort() {
+        return super.usingAnyFreePort();
+    }
+
+    /**
+     * Defines the environment for the launched appium server.
+     *
+     * @param environment A map of the environment variables to launch the
+     *     appium server with.
+     * @return A self reference.
+     */
+    @Override
+    public AppiumServiceBuilder withEnvironment(Map<String, String> environment) {
+        return super.withEnvironment(environment);
+    }
+
+    /**
+     * Configures the appium server to write log to the given file.
+     *
+     * @param logFile A file to write log to.
+     * @return A self reference.
+     */
+    public AppiumServiceBuilder withLogFile(File logFile) {
+        return super.withLogFile(logFile);
+    }
+
+    @Override
+    protected AppiumDriverLocalService createDriverService(File nodeJSExecutable, int nodeJSPort, ImmutableList<String> nodeArguments,
+                                                           ImmutableMap<String, String> nodeEnvironment) {
+        try {
+            return new AppiumDriverLocalService(ipAddress, nodeJSExecutable, nodeJSPort, nodeArguments, nodeEnvironment,
+                    timeToWaitForAppiumNodeIsStarted, timeUnitToWaitForAppiumNodeIsStarted);
```

Comment:
The naming seems excessively verbose.

`timeToWaitForAppiumNodeIsStarted, timeUnitToWaitForAppiumNodeIsStarted`

What about something shorter such as:

`startupTimeout, unit`